packages/data/src/redux-store/metadata/reducer.ts

```diff
@@ -13,15 +13,26 @@ import { selectorArgsToStateKey, onSubKey } from './utils';
 type Action =
 	| ReturnType< typeof import('./actions').startResolution >
 	| ReturnType< typeof import('./actions').finishResolution >
+	| ReturnType< typeof import('./actions').failResolution >
 	| ReturnType< typeof import('./actions').startResolutions >
 	| ReturnType< typeof import('./actions').finishResolutions >
+	| ReturnType< typeof import('./actions').failResolutions >
 	| ReturnType< typeof import('./actions').invalidateResolution >
 	| ReturnType< typeof import('./actions').invalidateResolutionForStore >
 	| ReturnType<
 			typeof import('./actions').invalidateResolutionForStoreSelector
 	  >;
 
-export type State = EquivalentKeyMap< unknown[] | unknown, boolean >;
+type StateKey = unknown[] | unknown;
+export enum Status {
```

Comment:
Done @kevin940726

Reply:
I mean instead of using `undefined` as an enum value, let's just map it to `IDLE` internally. 

```ts
enum StateKey {
  IDLE: 'IDLE',
  RESOLVING: 'RESOLVING',
  // ...
}

// In getResolutionState()
if ( ! resolutionState ) {
    return { status: StatusKey.IDLE }
}
```

So that we don't have to use a random undefined in a union type.

Reply:
I'd rather not do that, the reducer never returns that value. Now that I think about it more, it could add confusion to  `createResolversCacheMiddleware` where we access the raw state and not the value returned by `getResolutionState`. I reverted these types to simply reflect the underlying state without any augmenting – I'm still happy to keep discussing that separately from this PR.

Reply:
I don't feel strongly about this anyway 😆, just approved! Thank you for your hard work and patience! 🙇‍♂️